From 8d7a0174ba5ba7095c63c22cd7f40eddfcdd7fea Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Tue, 14 Oct 2014 16:44:57 +0800
Subject: [PATCH 0916/1543] MLK-9691 net: fec: call .pm_qos_remove_request()
 in .ndo_stop() callback

commit 3b1d089c50a7017aaeb733ae2f524b8ea0e87729 from
git://git.freescale.com/imx/linux-2.6-imx.git

Call .pm_qos_remove_request() in .ndo_stop() callback to avoid kernel
warning during enet open/close test.

Signed-off-by: Fugang Duan <B38611@freescale.com>
---
 drivers/net/ethernet/freescale/fec_main.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/net/ethernet/freescale/fec_main.c b/drivers/net/ethernet/freescale/fec_main.c
index c541f37..cc0619c 100644
--- a/drivers/net/ethernet/freescale/fec_main.c
+++ b/drivers/net/ethernet/freescale/fec_main.c
@@ -2776,6 +2776,7 @@ fec_enet_close(struct net_device *ndev)
 	fep->phy_dev = NULL;
 
 	fec_enet_clk_enable(ndev, false);
+	pm_qos_remove_request(&ndev->pm_qos_req);
 	pinctrl_pm_select_sleep_state(&fep->pdev->dev);
 	fec_enet_free_buffers(ndev);
 
-- 
1.7.5.4

